accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,550 @@
+importantly, this notation will also enable developers to intuitively understand
+compatibility relationships because they are expressed by naming, rather than by
+mapping tables. Yay!
+
+## Scenarios and User Experience
+
+### Vary implementation
+
+Ida is working on a Xamarin Forms application that supports Android, iOS, and
+Windows. Her application needs GPS information, but only a very limited set.
+Since there is no portable GPS API, she writes her own little abstraction
+library using multi-targeting.
+
+By doing so, she's able to encapsulate the GPS access without having to
+multi-target her entire application, just this one area.
+
+```C#
+public static class GpsLocation
+{
+    public static bool IsSupported
+    {
+        get
+        {
+#if ANDROID || IOS || WINDOWS
+            return true;
+#else
+            return false;
+#endif
+        }
+    }
+
+    public static (double Latitude, double Longitude) GetCoordinates()
+    {
+#if ANDROID
+        return AndroidAPI();
+#elif IOS
+        return AppleAPI();
+#elif WINDOWS
+        return WindowsAPI();
+#else
+        throw new PlatformNotSupportedException();
+#endif
+    }
+}
+```
+
+### Vary API surface
+
+Ada is a developer on SkiaSharp, a cross-platform 2D graphics API for .NET
+platforms based on Google's Skia Graphics Library. The project is already using
+multi-targeting to provide different implementations for different platform. To
+make it easier to use, she's adding a new `SkiaSharpImage` type, which
+represents a bitmap and is constructed via OS-provided data types. Ada uses
+`#if` to expose different constructors on different platforms:
+
+```C#
+public static class SkiaSharpImage
+{
+#if ANDROID
+    public SkiaSharpImage(Android.Media.Image nativeImage) { /* ... */  }
+#endif
+
+#if IOS
+    public SkiaSharpImage(NSImage nativeImage) { /* ... */ }
+#endif
+
+#if WINDOWS
+    public SkiaSharpImage(Windows.Media.BitmapImage nativeImage) { /* ... */ }
+#endif
+}
+```
+
+## Requirements
+
+### Goals
+
+* Use naming that aligns with product strategy
+* Merge .NET Core and .NET Standard into a single concept
+* Developers should be able to understand compatibility relationships without
+  having to consult a mapping table.
+* Provide compatibility with existing concepts and NuGet packages
+* It would be great to get this into .NET 5 Preview 1
+
+### Non-Goals
+
+* Replace TFMs or expand runtime identifiers (RIDs)
+* Support multi-targeting between OS versions
+
+## Design
+
+| TFM             | Compatible With                                            | Comments                          |
+|-----------------|------------------------------------------------------------|-----------------------------------|
+| net5.0          | net1..4 (with NU1701 warning)                              | No WinForms or WPF                |
+|                 | netcoreapp1..3.1 (warning when WinForms/WPF is referenced) |                                   |
```

Comment:
@terrajobst  - can you suggest the warning text for net5.0 -> netcoreapp3.1 w/ framework ref bringing in WPF?

Reply:
NU1703 draft text:
Package 'x 1.0.0' was restored using a targetFramework with a FrameworkReference in 'Microsoft.WindowsDesktop.App' instead of the project target framework '.NETCoreApp,Version=v5.0'. This package may not be fully compatible with your project.

```
  <data name="Log_ImportsFallbackWarning" xml:space="preserve">
    <value>Package '{0}' was restored using '{1}' instead of the project target framework '{2}'. This package may not be fully compatible with your project.</value>
  </data>
  <data name="Log_UsingWindowsDesktopFrameworkRefWarning" xml:space="preserve">
    <value>Package '{0}' was restored using a targetFramework with a FrameworkReference in '{2}' instead of the project target framework '{1}'. This package may not be fully compatible with your project.</value>
  </data>
```

Reply:
@rrelyea 

I have to admit that I don't understand this message 😄 

How about this:

> Package 'x 1.0.0' was restored using 'netcoreapp3.1' with a framework reference to 'Microsoft.WindowsDesktop.App', which isn't available on all operating systems, which means this package might not be fully compatible with your project. To limit your project to Windows, change your target framework from `net6.0` to `net6.0-windows`.